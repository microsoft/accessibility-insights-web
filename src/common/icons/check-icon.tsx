// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

import { NamedFC } from '../react/named-fc';

export const CheckIcon = NamedFC('CheckIcon', () => (
    <span className="check-container">
        <svg
            width="16"
            height="16"
            viewBox="0 0 16 16"
            fill="none"
            xmlns="http://www.w3.org/2000/svg"
            role="img"
            aria-hidden="true"
        >
            <circle cx="8" cy="8" r="8" fill="#228722" />
            <path
                fillRule="evenodd"
                clipRule="evenodd"
                d="M6.05904 11.1417L6.0616 11.1442C6.10737 11.19 6.15668 11.23 6.20867 11.2643C6.57256 11.5046 7.06707 11.4646 7.38742 11.1442C7.38861 11.143 7.38982 11.1418 7.391 11.1406L11.9312 6.60041C12.2974 6.23427 12.2974 5.64071 11.9312 5.27457C11.5651 4.90848 10.9715 4.90848 10.6054 5.27457L6.72452 9.15545L5.60041 8.03134C5.2343 7.66524 4.64071 7.66524 4.27459 8.03134C3.90847 8.39747 3.90847 8.99104 4.27459 9.35718L6.05904 11.1417Z"
                fill="var(--neutral-0)"
            />
        </svg>
    </span>
));

export const CheckIconInverted = NamedFC('CheckIconInverted', () => (
    <span className="check-container">
        <svg
            width="16"
            height="16"
            viewBox="0 0 16 16"
            fill="none"
            xmlns="http://www.w3.org/2000/svg"
            role="img"
            aria-hidden="true"
        >
            <path
                fillRule="evenodd"
                clipRule="evenodd"
                d="M8 16C12.4183 16 16 12.4183 16 8C16 3.58172 12.4183 0 8 0C3.58172 0 0 3.58172 0 8C0 12.4183 3.58172 16 8 16ZM6.0616 11.1442L6.05904 11.1417L4.27459 9.35718C3.90847 8.99104 3.90847 8.39747 4.27459 8.03134C4.64071 7.66524 5.2343 7.66524 5.60041 8.03134L6.72452 9.15545L10.6054 5.27457C10.9715 4.90848 11.5651 4.90848 11.9312 5.27457C12.2974 5.64071 12.2974 6.23427 11.9312 6.60041L7.391 11.1406L7.38742 11.1442C7.06707 11.4646 6.57256 11.5046 6.20867 11.2643C6.15668 11.23 6.10737 11.19 6.0616 11.1442Z"
                fill="white"
            />
        </svg>
    </span>
));
