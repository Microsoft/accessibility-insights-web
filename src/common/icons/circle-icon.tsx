// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

import { NamedFC } from '../react/named-sfc';

export const CircleIcon = NamedFC('CircleIcon', () => <span className="check-container" />);
