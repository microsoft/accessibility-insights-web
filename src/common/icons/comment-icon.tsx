// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

import { NamedFC } from '../react/named-fc';

const d =
    'M16 11.2891V0.289062H-7.62939e-06V11.2891H1.99999V15L5.71093 11.2891H16ZM1 1.28906H15L15 10.2891H5.28906L3 ' +
    '12.5781V10.2891H1V1.28906ZM9.99999 6.28906H8.99999V4.28906H11V6.28906C11 6.42969 10.974 6.55859 10.9219 6.67969C10.8698 6.80078 ' +
    '10.7982 6.90625 10.707 6.99609C10.6159 7.08594 10.5104 7.16016 10.3906 7.21094C10.2708 7.26172 10.1406 7.28906 9.99999 ' +
    '7.28906V6.28906ZM4.99999 6.28906H5.99999V7.28906C6.14062 7.28906 6.27083 7.26172 6.39062 7.21094C6.51041 7.16016 6.61588 ' +
    '7.08594 6.70702 6.99609C6.79817 6.90625 6.86978 6.80078 6.92187 6.67969C6.97395 6.55859 6.99999 6.42969 6.99999 ' +
    '6.28906V4.28906H4.99999V6.28906Z';

export const CommentIcon = NamedFC('CommentIcon', () => (
    <svg
        width="17"
        height="16"
        viewBox="0 0 16 15"
        fill="none"
        xmlns="http://www.w3.org/2000/svg"
        role="img"
        aria-hidden="true"
    >
        <path fillRule="evenodd" clipRule="evenodd" d={d} fill="#737373" />
    </svg>
));
