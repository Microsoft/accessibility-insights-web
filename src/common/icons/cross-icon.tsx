// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

import { NamedSFC } from '../react/named-sfc';

export const CrossIcon = NamedSFC('CrossIcon', () => (
    <span className="check-container">
        <span className="cross-icon" />
    </span>
));
