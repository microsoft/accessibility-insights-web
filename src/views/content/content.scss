// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
@import '../../common/icons/icon.scss';
@import '../../common/styles/colors.scss';
@import '../../common/styles/fonts.scss';

.content-container {
    display: flex;

    .content-left,
    .content-right {
        flex-grow: 1;
        max-width: 150px;
    }
}

.content-header {
    @include text-style-title-l;

    margin-left: 10px;
    margin-right: 10px;
}

.content {
    flex-grow: 10;
    max-width: 1120px;
    font-family: $font-family;
    font-size: 14px;
    margin: 10px;
    word-break: break-word;

    table {
        font-family: $font-family;
        font-size: 14px;
    }

    h1 {
        @include text-style-title-l;
    }

    h2 {
        font-size: 21px;
        font-weight: 700;
        letter-spacing: -0.02em;
        line-height: 32px;
    }

    h3 {
        font-size: 14px;
    }

    h4 {
        font-weight: 400;
        font-size: 14px;
        margin-bottom: 4px;
    }

    h5 {
        font-weight: 700;
        font-size: 14px;
        margin-bottom: 4px;
    }

    a {
        font-weight: 400;
    }

    ol {
        -webkit-padding-start: 16px; // Chrome versions before 87 use non-standard name: https://caniuse.com/mdn-css_properties_padding-inline-start
        padding-inline-start: 16px;

        li ul li {
            list-style-type: disc;
        }
    }

    .content-hyperlinks {
        .insights-link {
            line-height: 20px;
            margin-bottom: 6px;
            display: block;
        }

        .content-inline {
            line-height: 20px;
            margin-bottom: 6px;

            .insights-link {
                display: inline;
            }
        }
    }

    .code-example {
        .code-example-code {
            font-family: $code-font-family;
            margin-block-start: 1em;
            margin-block-end: 1em;
        }

        .code-example-title {
            h4 {
                display: inline;
                margin-block-start: 0;
            }
        }
    }

    .columns {
        margin-top: 40px;
        @media screen and (width >= 800px) {
            display: flex;
        }
    }

    .column {
        padding: 10px;
        flex-grow: 1;
        flex-basis: 0;
        max-width: 520px;
    }

    .column + .column {
        @media screen and (width >= 800px) {
            margin-left: 80px;
        }
    }

    $icon-size: 16px;
    $do-color: $positive-outcome;
    $dont-color: $negative-outcome;

    .insights-code {
        font-family: $code-font-family;
    }

    .pass-fail-grid {
        display: grid;
        grid-template-areas:
            'fail-section pass-section'
            'fail-example pass-example';
        border: 1px solid $neutral-60;
        word-break: break-word;
    }

    .fail-section,
    .fail-example,
    .pass-section,
    .pass-example {
        padding: 15px;
    }

    .fail-example,
    .pass-example {
        border-top: 1px solid $neutral-60;
    }

    .fail-section,
    .fail-example {
        border-right: 1px solid $neutral-60;
    }

    .fail-section {
        grid-area: fail-section;
    }

    .fail-example {
        grid-area: fail-example;
    }

    .pass-section {
        grid-area: pass-section;
    }

    .pass-example {
        grid-area: pass-example;
    }

    .do-header,
    .pass-header,
    .dont-header,
    .fail-header {
        width: 100%;
    }

    .do-header,
    .dont-header {
        .check-container {
            margin-top: 9px;
            float: right;
        }

        h2 {
            display: inline;
        }
    }

    .pass-header,
    .fail-header {
        h3 {
            display: inline;
        }
    }

    .do-section,
    .pass-section,
    .pass-example {
        .highlight {
            color: $do-color;
            font-weight: bold;
        }
    }

    .dont-section,
    .fail-section,
    .fail-example {
        .highlight {
            color: $dont-color;
            font-weight: bold;
        }
    }

    .do-header {
        @include check-icon-styles($icon-size, 0, $do-color);

        border-bottom: 2px solid $do-color;
    }

    .dont-header {
        @include cross-icon-styles($icon-size, 0, $dont-color);

        border-bottom: 2px solid $dont-color;
    }

    .pass-header {
        @include check-icon-styles(12px, 0, $do-color);
    }

    .fail-header {
        @include cross-icon-styles(12px, 0, $dont-color);
    }

    .table {
        list-style: none;
        margin-top: 12px;
        display: table;
        border-collapse: separate;
        border-spacing: 5px;
        padding-left: 18px;
    }

    .accessibility-problems-list {
        background: $neutral-6;
        list-style-type: disc;
    }

    .landmark-table {
        border-spacing: 5px;

        @media screen and (width >= 600px) {
            padding-left: 32px;
        }

        .landmark-table-column {
            width: 24%;
        }

        .landmark-role {
            text-align: center;
            border-style: dashed;
            border-width: 3px;
        }

        td,
        th {
            padding: 5px;
            vertical-align: text-top;
        }

        th {
            background: $neutral-100;
            color: $neutral-0;
            font-weight: 400;
        }

        td {
            background: $neutral-10;
            color: $neutral-100;
        }

        .contentinfo-landmark {
            border-color: $landmark-contentinfo;
        }

        .main-landmark {
            border-color: $landmark-main;
        }

        .complementary-landmark {
            border-color: $landmark-complementary;
        }

        .banner-landmark {
            border-color: $landmark-banner;
        }

        .region-landmark {
            border-color: $landmark-region;
        }

        .navigation-landmark {
            border-color: $landmark-navigation;
        }

        .search-landmark {
            border-color: $landmark-search;
        }

        .form-landmark {
            border-color: $landmark-form;
        }
    }

    .landmarks-legend {
        color: $neutral-100;
        font-size: 14px;
        display: table-cell;
        width: 110px;
        text-align: center;
        font-weight: 400;
        padding: 4px;
        vertical-align: middle;
        border-style: dashed;
        border-width: 3px;

        &.contentinfo-landmark {
            border-color: $landmark-contentinfo;
        }

        &.main-landmark {
            border-color: $landmark-main;
        }

        &.complementary-landmark {
            border-color: $landmark-complementary;
        }

        &.banner-landmark {
            border-color: $landmark-banner;
        }

        &.region-landmark {
            border-color: $landmark-region;
        }

        &.navigation-landmark {
            border-color: $landmark-navigation;
        }

        &.search-landmark {
            border-color: $landmark-search;
        }

        &.form-landmark {
            border-color: $landmark-form;
        }
    }
}
