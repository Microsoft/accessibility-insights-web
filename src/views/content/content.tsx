// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

import { NamedSFC } from '../../common/react/named-sfc';
import { ContentPageDeps, ContentProvider, ContentReference } from './content-page';
import { Page } from '../page/page';

export type ContentDeps = { contentProvider: ContentProvider } & ContentPageDeps;

export type ContentProps = { deps: ContentDeps; reference: ContentReference };

export const Content = NamedSFC<ContentProps>('Content', ({ deps, reference }) => {
    const { contentProvider } = deps;
    const ContentPage = contentProvider.contentFromReference(reference);
    return (
        <Page>
            <div className="content-container">
                <div className="content-left" />
                <div className="content">
                    <ContentPage deps={deps} options={{ setPageTitle: true }} />
                </div>
                <div className="content-right" />
            </div>
        </Page>
    );
});
