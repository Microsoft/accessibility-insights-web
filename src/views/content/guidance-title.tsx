// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';
import Helmet from 'react-helmet';
import { NamedSFC } from '../../common/react/named-sfc';
import { productName } from '../../content/strings/application';

export type GuidanceTitleProps = {
    name: string;
};

export const GuidanceTitle = NamedSFC<GuidanceTitleProps>('GuidanceTitle', ({ name }) => (
    <>
        <Helmet>
            <title>
                Guidance for {name} - {productName}
            </title>
        </Helmet>
        <h1>{name}</h1>
    </>
));
