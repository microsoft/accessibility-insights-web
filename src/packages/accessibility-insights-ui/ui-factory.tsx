// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { CheckIcon } from 'common/icons/check-icon';
import { CrossIcon } from 'common/icons/cross-icon';
import { NamedFC } from 'common/react/named-fc';
import { NarrowModeThresholds } from 'common/narrow-mode-thresholds';
import * as React from 'react';
import { ContentView, ContentViewDeps } from 'views/content/content-view';
import { GuidanceTitle } from 'views/content/guidance-title';
import { createMarkup, MarkupDeps } from 'views/content/markup';

type UIOptions = {
    applicationTitle: string;
    setPageTitle: boolean;
    getNarrowModeThresholds: () => NarrowModeThresholds;
};

const nullCreator = () => {};

export const contentViewFactory = ({ applicationTitle, getNarrowModeThresholds }: UIOptions) => {
    const contentViewDeps: ContentViewDeps = {
        textContent: {
            applicationTitle,
        },
        storeActionMessageCreator: {
            getAllStates: nullCreator,
        },
        storesHub: null,
        storesActionCreator: null,
        getNarrowModeThresholds,
    };

    return NamedFC('ContentView', props => (
        <ContentView deps={contentViewDeps}>{props.children}</ContentView>
    ));
};

export const markupFactory = (options: UIOptions) => {
    const { applicationTitle } = options;
    const markupDeps: MarkupDeps = {
        textContent: {
            applicationTitle,
        },
        contentActionMessageCreator: {
            openContentHyperLink: nullCreator,
        },
    };
    const { Include, ...Markup } = createMarkup(markupDeps, options);
    return Markup;
};

export const UIFactory = (options: UIOptions) => {
    return {
        ContentView: contentViewFactory(options),
        GuidanceTitle,
        Markup: markupFactory(options),
        CheckIcon,
        CrossIcon,
    };
};
