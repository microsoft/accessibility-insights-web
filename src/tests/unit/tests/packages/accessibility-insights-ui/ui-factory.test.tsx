// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { CheckIcon } from 'common/icons/check-icon';
import { CrossIcon } from 'common/icons/cross-icon';
import { shallow } from 'enzyme';
import { UIFactory } from 'packages/accessibility-insights-ui/ui-factory';
import * as React from 'react';
import { GuidanceTitle } from 'views/content/guidance-title';

describe('UIFactory', () => {
    const applicationTitle = 'THE_APPLICATION_TITLE';
    const narrowModeThresholds = {
        collapseHeaderAndNavThreshold: 600,
        collapseCommandBarThreshold: 960,
    };
    const options = {
        applicationTitle,
        setPageTitle: true,
        getNarrowModeThresholds: () => narrowModeThresholds,
    };
    const ui = UIFactory(options);

    it('exports CheckIcon', () => {
        expect(ui.CheckIcon).toBe(CheckIcon);
    });
    it('exports CrossIcon', () => {
        expect(ui.CrossIcon).toBe(CrossIcon);
    });
    it('exports GuidanceTitle', () => {
        expect(ui.GuidanceTitle).toBe(GuidanceTitle);
    });
    describe('exports Markup', () => {
        const markup = ui.Markup;

        const exports = [
            'Code',
            'CodeExample',
            'Column',
            'Columns',
            'Do',
            'Dont',
            'Emphasis',
            'Fail',
            'Highlight',
            'HyperLink',
            'Inline',
            'LandmarkLegend',
            'Links',
            'Pass',
            'PassFail',
            'ProblemList',
            'Table',
            'Tag',
            'Term',
            'Title',
        ];

        it.each(exports)('exports %s', exp => {
            expect(markup).toHaveProperty(exp);
        });

        it('does not export Include', () => {
            expect(markup).not.toHaveProperty('Include');
        });

        it('has correct options', () => {
            const opts = (markup as any).options;
            expect(opts.applicationTitle).toEqual(applicationTitle);
            expect(opts.setPageTitle).toEqual(true);
            expect(opts.getNarrowModeThresholds()).toEqual(narrowModeThresholds);
        });
    });

    it('exports ContentView', () => {
        const wrapper = shallow(<ui.ContentView>CONTENTS</ui.ContentView>);
        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
