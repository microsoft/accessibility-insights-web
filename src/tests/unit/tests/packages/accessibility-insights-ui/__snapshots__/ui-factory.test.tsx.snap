// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UIFactory exports ContentView 1`] = `
<Content
  deps={
    Object {
      "getNarrowModeThresholds": [Function],
      "storeActionMessageCreator": Object {
        "getAllStates": [Function],
      },
      "storesActionCreator": null,
      "storesHub": null,
      "textContent": Object {
        "applicationTitle": "THE_APPLICATION_TITLE",
      },
    }
  }
>
  CONTENTS
</Content>
`;
